manager/src/main/java/org/openremote/manager/map/MapService.java

```diff
@@ -442,6 +461,36 @@ public byte[] getMapTile(int zoom, int column, int row) {
         }
     }
 
+    public URI getExternalMapTileUri(int zoom, int column, int row) {
+        final ObjectNode settings = (metadata.isValid() && !mapConfig.isEmpty()) 
+                ? mapConfig.deepCopy()
+                : mapConfig.objectNode();
+
+        final String source = settings
+            .get("sources")
+            .get("external")
+            .get("url")
+            .textValue();
+
+        if (!source.contains("{secret}")) {
+            return null;
+        }
+
+        final String secret = settings
+            .get("sources")
+            .get("external")
+            .get("secret")
+            .textValue();
```

Comment:
might want to retrieve from an encrypted file instead.

Reply:
The tile server tokens are usually public, so this can be ignored.

![image](https://github.com/user-attachments/assets/c81603a9-68a9-44bb-bce0-3a47e36052f1)